exercises/matthew_cumber/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <iostream>
+#include <iomanip>
+#include <algorithm>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <ctype.h>
+#include "UniqueWord.h"
+
+// Default path for output file reading files from
+std::string const PATH = "/Users/Matt/STFC Placement/C++ Introduction/newstarter/exercises/matthew_cumber/ex01_basics/";
+
+/** Tests if a word has already been found the increments it's count
+ * @param uniqueWords The vector of existing unique words found
+ * @param word The word that is being tested
+ * @return A boolean, true if a word already exists, false otherwise
+ */
+bool inUniqueWords(std::vector<UniqueWord> &uniqueWords,std::string word)
+{
+  // Loop unique words, if word exists increment it's count
+  for(auto &w : uniqueWords)
+  {
+    if(word == w.getWord())
+    {
+      w.incrementCount();
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/** Tests if a letter is one of the predefined punctuation characters
+ * @param letter The letter which is being tested
+ * @return A boolean, true if a letter is a punctuation character, false otherwise
+ */
+bool isPunctuation(const char letter)
+{
+  for(int i = 0; i < PUNCTUATION_COUNT; ++i)
+  {
+    if(letter == PUNCTUATION[i]) return true;
+  }
+
+  return false;
+}
+
+/** Writes the unique words to an output file "UniqueWords.txt"
+ * @param uniqueWords The vector of unique words to be written to the file
+ * @return no return value
+ */
+void writeUniqueWordsToFile(std::vector<UniqueWord> const uniqueWords)
```

Comment:
You should pass the parameter by reference for better performance. Also as a matter of style\consistency I suggest moving the const should be before the std::vector (but I think that doesn't have any functional difference):

ie const std::vector<UniqueWord>& uniqueWords